src/tests/integration/test-data/foundry/src/cse/lemmas.k

```diff
@@ -0,0 +1,33 @@
+requires "evm.md"
+requires "foundry.md"
+
+module CSE-LEMMAS
+    imports BOOL
+    imports FOUNDRY
+    imports INFINITE-GAS
+    imports INT-SYMBOLIC
+    imports MAP-SYMBOLIC
+    imports SET-SYMBOLIC
+
+    // xor in terms of -Int
+    rule X xorInt maxUInt256 => maxUInt256 -Int X
+      requires #rangeUInt ( 256 , X )
+      [simplification]
+
+    // for-loop chop
+    rule chop ( ( X:Int +Int Y:Int ) ) ==Int 0 => X ==Int pow256 -Int (Y modInt pow256)
+      requires #rangeUInt(256, X) andBool 0 <=Int Y
+      [simplification, concrete(Y)]
+
+    // Set equality needed for discharging `#Not ( #Exists ( ... )` on `<accessedAccounts>` unification
+    rule { S1:Set #Equals S2:Set |Set SetItem ( X ) } =>
+         { X in S1 } #And
+         ( { S2 #Equals S1 } #Or { S2 #Equals S1 -Set SetItem ( X ) } )
+         [simplification]
+
+endmodule
```

Comment:
These  seem like lemmas we can get upstreamed into KEVM with appropriate tests (the motivating expressions that caused these simplifications to be needed).

Reply:
The first two, yes, the third one is needed until the KCFG-to-KCFG transformation that summarises the behaviour in one-step is done.